src/InfoBoxes/Content/Radio.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "InfoBoxes/Content/Radio.hpp"
+#include "InfoBoxes/Data.hpp"
+#include "Interface.hpp"
+#include "Units/Units.hpp"
+#include "Formatter/UserUnits.hpp"
+#include "Language/Language.hpp"
+
+#include <tchar.h>
+
+void
+UpdateInfoBoxFrequency(InfoBoxData & data, const RadioFrequency & freq, const TCHAR * freq_name)
+{
+  if(freq.IsDefined()) {
+    TCHAR buffer[32];
+    freq.Format(buffer, 32);
+    data.SetValue(buffer);
+    data.SetComment(freq_name);
```

Comment:
This will crash if `freq_name==nullptr`. The parameter is not documented, and is not documented that `nullptr` is an invalid value. In another part of your code, `freq_name==nullptr` is a legal parameter. So this really needs clarification.

Reply:
Changed parameter to `const StaticString<32> &`. This does not allow nullptr, and matches the type that is read from ComputerSettings.
With the changes indicated above, nullptr is not a valid option in this part of the code.

Reply:
That's a bad API decision because it is extremely narrow.

Reply:
Thanks for the response. I have expanded the length to 64, but if that too isn't satisfactory I kindly ask for some insights.
My reasoning is this:
1. I don't want to use a char *, because I can't guarantee that the name is still in memory when I change task/waypoints after setting the frequency. So I need a copy.
2. The CUP file format (for example) does not define a maximum length for waypoint names.
3. I think using a tstring is not a good option, since I am not sure that ComputerSettings would remain trivial. It would also imply copying from this tstring to the StaticString<32> defined in InfoBoxData.

Thus, since the only usage so far is the InfoBox, I decided that StaticString<32> would be best for now.

If any of the points above are invalid, please let me know.

Reply:
About 1.: Your `StaticString<32>&` just as well doesn't guarantee that the name is still in memory.
2. and 3. are valid, but I have no idea how any of this (including 1.) is relevant for this API design.

Reply:
Thanks for taking the time to respond.

I will look into 1, even though it might not be relevant.

I think I don't understand what you mean by "API decision" and "narrow" in this context. The function I think we are discussing is a static function, which in my view exists only because I don't want to repeat it's contents two times. I don't think it is part of any API.

I thought with "narrow" you meant the maximum size of 32, but I'm starting to doubt that now. Can you please clarify?

Reply:
"Narrow" means the choice of types for this parameter is very narrow. Only `const StaticString<32>&` is allowed, even though there is no technical reason for this limitation. Previously, you had `const TCHAR *` which was just fine, and I don't know why you changed it. Any `StaticString` is implicitly convertibly to `const TCHAR *` and that without any overhead; but you can convert `const TCHAR *` to a `StaticString` only by allocating memory and copying data.

Reply:
Okay, thanks for clarifying!

Reply:
I've changed the parameter back to `const TCHAR *`. I believe there are still copies made during assignment in `ComputerSettings` and assignment of the `StaticString` inside `InfoBoxData`, but I currently don't see how to avoid those while keeping `ComputerSettings` trivial.